main/test/ca/uwaterloo/flix/util/TestChain.scala

```diff
@@ -54,4 +54,49 @@ class TestChain extends AnyFunSuite {
     val chain = Chain.concat(chains)
     assert(chain.toList == List(1, 2, 3, 4))
   }
+
+  test("TestEq.01") {
+    assertResult(Chain.empty)(Chain.empty)
+  }
+
+  test("TestEq.02") {
+    val c1 = Chain(1, 2, 3, 4, 5)
+    assertResult(c1)(c1)
+  }
+
+  test("TestEq.03") {
+    val c1 = Chain(1, 2, 3, 4, 5)
+    val c2 = Chain(1, 2, 3, 4, 5)
+    assertResult(c1)(c2)
+  }
+
+  test("TestEq.04") {
+    val c1 = Chain(1) ++ Chain(2)
+    val c2 = Chain(1, 2)
+    assertResult(c1)(c2)
+  }
+
+  test("TestEq.05") {
+    val c1 = Chain.concat(Seq(Chain(1), Chain(2), Chain(3), Chain(4), Chain(5)))
+    val c2 = Chain(1, 2, 3, 4, 5)
+    assertResult(c1)(c2)
+  }
+
+  test("TestEq.06") {
+    val c1 = Chain(1, 2, 3, 4, 5)
+    val c2 = Chain(1, 1, 3, 4, 5)
+    assert(c1 != c2)
+  }
+
+  test("TestEq.07") {
+    val c1 = Chain(1) ++ Chain(3)
+    val c2 = Chain(1, 2)
+    assert(c1 != c2)
+  }
+
+  test("TestEq.08") {
```

Comment:
I notice that none of these tests check for equality between chains that are of very different shape.

Reply:
Like e.g. a purely left-leaning versus purely right-leaning chain with the same elements.

Reply:
Ah yes, I'll look into it.